### PR TITLE
Rotations

### PR DESCRIPTION
**Pull request by pham-van-tuy**
fix_rotation

### CODE REVIEW

File: app/src/main/java/com/example/learn09082021v001/DataInput.kt [hunk 1]
```diff
@@ -1,56 +1,72 @@
...
-import android.widget.Toast
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
 import kotlinx.android.synthetic.main.activity_data_input.*
 
 
 class DataInput : AppCompatActivity() {
+    override fun onSaveInstanceState(savedInstanceState: Bundle) {
+        super.onSaveInstanceState(savedInstanceState)
+        savedInstanceState.putString("nameandroid", edtNameAndroid.text.toString())
```
Comment (thuypt) [inline app/src/main/java/com/example/learn09082021v001/DataInput.kt:17]: create a constant class to handle all of these keys, don't use hard code
"nameandroid"
"scoreandroi"
"nameios"

File: app/src/main/java/com/example/learn09082021v001/DataInput.kt [hunk 2]
```diff
@@ -1,56 +1,72 @@
...
+        btnDone.setOnClickListener {
+            scoreOOP.NameInputandroid = edtNameAndroid.text.toString()
+            scoreOOP.NameInPutIOS = edtNameAndroid.text.toString()
             val intent = Intent(this, MainActivity::class.java)
-            intent.putExtra("AndroidName", edtNameAndroid.text.toString())
-            intent.putExtra("IOSName", edtNameIOS.text.toString())
-            intent.putExtra("AndroidScore", tvScoreAndroid.text.toString())
-            intent.putExtra("IOSScore", tvScoreIOS.text.toString())
-            startActivity(intent)
+            intent.putExtra("nameandroidSend", edtNameAndroid.text.toString())
```
Comment (thuypt) [inline app/src/main/java/com/example/learn09082021v001/DataInput.kt:60]: same here, don't hardcode string value, especially value are reuse able

File: app/src/main/java/com/example/learn09082021v001/MainActivity.kt [hunk 3]
```diff
@@ -1,39 +1,66 @@
...
+            } else { dataclear () }
         }
 
+
     }
+    fun datacheck () {
+        if (tvViewNameAndroid.text.isNotEmpty() || tvViewNameIOS.text.isNotEmpty()
+            || tvViewScoreAndroid.text.isNotEmpty() || tvViewScoreIOS.text.isNotEmpty()) {
+            btnStartReset.setText("RESET") } else btnStartReset.setText("START") }
+    fun dataclear (){
```
Comment (thuypt) [inline app/src/main/java/com/example/learn09082021v001/MainActivity.kt:60]: the function name should be a verb, like `clearData`, same with `datacheck`

File: app/src/main/java/com/example/learn09082021v001/MainActivity.kt [hunk 4]
```diff
@@ -1,39 +1,66 @@
...
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (data != null && resultCode == RESULT_OK) {
+            tvViewNameAndroid.setText(data.getStringExtra("nameandroidSend"))
+            tvViewNameIOS.setText(data.getStringExtra("nameiosSend"))
+            tvViewScoreAndroid.setText(data.getStringExtra("scoreandroidSend"))
+            tvViewScoreIOS.setText(data.getStringExtra("scoreiosSend"))
+            datacheck()
```
Comment (thuypt) [inline app/src/main/java/com/example/learn09082021v001/MainActivity.kt:39]: call `datacheck` here is redundant. If necessary to validate data, it should be called before set data to views

Comment (pham-van-tuy) [reply app/src/main/java/com/example/learn09082021v001/MainActivity.kt:39]: datacheck funtion : it is check data empty or data not empty : will settext btn >> START OR RESET

**Pull request merged**